src/js/utils/element-utils.js

```diff
@@ -59,11 +63,69 @@ export function getOrInsertCSSRule(styleParent, selectorText) {
     return {
       style: {
         setProperty: () => {},
-        removeProperty: () => {}
-      }
+        removeProperty: () => {},
+      },
     };
   }
 
   style.sheet.insertRule(`${selectorText}{}`, style.sheet.cssRules.length);
   return style.sheet.cssRules[style.sheet.cssRules.length - 1];
 }
+
+/**
+ * Gets the number represented by the attribute
+ * @param {HTMLElement} el
+ * @param {string} attrName
+ * @returns {number | undefined} Will return undefined if no attribute set
+ */
+export function getNumericAttr(el, attrName) {
+  const attrVal = el.getAttribute(attrName);
+  return attrVal != null ? +attrVal : undefined;
+}
+
+/**
+ * @param {HTMLElement} el
+ * @param {string} attrName
+ * @param {number} value
+ */
+export function setNumericAttr(el, attrName, value) {
+  // avoid setting a value that hasn't changed
+  if (getNumericAttr(el, attrName) == value) return;
+
+  // also handles undefined
+  if (value == null) {
```

Comment:
Also need to handle `NaN` here somehow.

Reply:
As it currently is implemented `NaN` is handled implicitly as any other number. Setting `NaN` stores it in the DOM attribute as a string and using the getter retrieves it as a real `NaN` value.

Would you prefer it to be treated the same as `null` and `undefined`? that is: if someone sets the prop to `NaN` the attribute is removed from the element, and absence of the attribute on the getter side would return `NaN`? I can see the benefit of narrowing the prop type from `number | undefined` to simply `number` as `typeof NaN === 'number'`. As long as that makes sense from a consumers perspective i.e. the value can never have a state of "undefined" it is always a value, with the "default" value being NaN.

Reply:
If I use this component in isolation, initialise it in a web page and access the prop `element.mediaDuration` and it returns `NaN` (because no value has been set), I think that is counter-intuitive from a developers perspective? I think I would expect a default like `0` (if it's important to maintain the purity of the return type) or `undefined` to signal that a value as never been set, so there is no value to return.